### PR TITLE
Dim title bar when unfocused

### LINKED ISSUE #322: Title bar text should be dimmed when window is not focused
Screenshot 2023-01-24 at 19 58 52

Guidelines: https://learn.microsoft.com/en-gb/windows/apps/develop/title-bar?tabs=wasdk#dos-and-donts
> Do make it obvious when your window is active or inactive. At a minimum, change the color of the text, icons, and buttons in your title bar.

### PR DESCRIPTION
**Pull request by JaiganeshKumaran**
Fixes #322.

### CODE REVIEW

**Comment by BreeceW** [comment]
I would use a VisualStateGroup to toggle the title bar colors. That way you have the styling done in XAML and your ThemeResources would automatically update. You could change the Brush into a bool (for active/not active) and use a StateTrigger to bind to the bool to toggle between them.

File: AmbientSounds.sln [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿Microsoft Visual Studio Solution File, Format Version 12.00
+﻿
```
Comment (dpaulino) [inline AmbientSounds.sln:1]: Please revert the changes in the sln file

File: src/AmbientSounds.Uwp/Controls/TitleBarLogo.xaml [hunk 2]
```diff
@@ -2,14 +2,16 @@
     x:Class="AmbientSounds.Controls.TitleBarLogo"
     xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
-    xmlns:strings="using:AmbientSounds.Strings">
+    xmlns:strings="using:AmbientSounds.Strings"
+    ActualThemeChanged="UserControl_ActualThemeChanged">
```
Comment (dpaulino) [inline src/AmbientSounds.Uwp/Controls/TitleBarLogo.xaml:6]: nit: use `OnEventRaised` method name styling

```suggestion
    ActualThemeChanged="OnActualThemeChanged">
```

File: src/AmbientSounds.Uwp/Controls/TitleBarLogo.xaml.cs [hunk 3]
```diff
@@ -12,14 +14,44 @@ public sealed partial class TitleBarLogo : UserControl
...
+    public Brush DisplayTextForegroundBrush
+    {
+        get => (Brush)GetValue(DisplayTextForegroundBrushProperty);
+        set => SetValue(DisplayTextForegroundBrushProperty, value);
+    }
+
+    private void SetDisplayTextForeground(bool isWindowFocused)
+    {
+        DisplayTextForegroundBrush = App.Current.Resources[isWindowFocused ?
+            "TextFillColorPrimaryBrush" : "TextFillColorTertiaryBrush"] as Brush;
```
Comment (dpaulino) [inline src/AmbientSounds.Uwp/Controls/TitleBarLogo.xaml.cs:45]: As mentioned in another comment, is it possible to re-implement this pr using visual states so we get the theme support automatically?

Comment (JaiganeshKumaran) [reply src/AmbientSounds.Uwp/Controls/TitleBarLogo.xaml.cs:45]: Sure.

**Comment by dpaulino** [comment]
@JaiganeshKumaran are you able to update the pr with the requested changes?

**Comment by JaiganeshKumaran** [comment]
@dpaulino I've updated the pull request now, please merge.

File: src/AmbientSounds.Uwp/Controls/TitleBarLogo.xaml.cs [hunk 4]
```diff
@@ -12,9 +13,22 @@ public sealed partial class TitleBarLogo : UserControl
...
+    public static readonly DependencyProperty IsWindowFocusedProperty =
+        DependencyProperty.Register("IsWindowFocused", typeof(bool), typeof(TitleBarLogo), new PropertyMetadata(false));
+
     public TitleBarLogo()
     {
         this.InitializeComponent();
+        Window.Current.Activated += (_, e) =>
+        {
+            IsWindowFocused = e.WindowActivationState != CoreWindowActivationState.Deactivated;
+        };
```
Comment (dpaulino) [inline src/AmbientSounds.Uwp/Controls/TitleBarLogo.xaml.cs:25]: This leads to a memory leak. Repro:
1. Launch ambie
2. navigate to screenshots page
3. press back button to return to home page
4. Observe that this event handler gets called twice (because it was registered twice)

To fix this, please do the following
1. Create event handlers for this control's `Loaded` and `Unloaded` events
2. Create a dedicated method that handles the window activated event
3. Using the dedicated method, add a `Windows.Current.Activated` registration to the `OnLoaded` event handler.
4. Add a unregister line in the `OnUnloaded` event handler

Once you've made the changes, run the repro steps from earlier. There shouldn't be a duplicate registration anymore.

**Review by dpaulino — CHANGES_REQUESTED** [review]

File: src/AmbientSounds.Uwp/Controls/TitleBarLogo.xaml.cs (hunk 4 above)
Comment (JaiganeshKumaran) [reply src/AmbientSounds.Uwp/Controls/TitleBarLogo.xaml.cs]: Done.

**Review by dpaulino — APPROVED** [review]
